application/clicommands/ListCommand.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/* Icinga Reporting | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Reporting\Clicommands;
+
+use Icinga\Module\Reporting\Cli\Command;
+use Icinga\Module\Reporting\Model;
+use InvalidArgumentException;
+use ipl\Stdlib\Filter;
+
+class ListCommand extends Command
+{
+    /**
+     * List reports
+     *
+     * USAGE
+     *
+     *     icingacli reporting list [OPTIONS]
+     *
+     * OPTIONS
+     *
+     * --sort=<id|name|author>
+     *     Sort the reports by the given column. Defaults to id.
+     *
+     * --direction=<asc|desc>
+     *     Sort the reports by the specified sort column in ascending or descending order. Defaults to asc.
+     *
+     * --filter=<name>
+     *     Filter the reports by the specified report name. Performs a wildcard search if the value contains asterisks.
+     *
+     * EXAMPLES
+     *
+     * Sort the reports by name:
+     *     icingacli reporting list --sort=name
+     *
+     * Sort the reports by author in descending order:
+     *     icingacli reporting list --sort=author --direction=DESC
+     *
+     * Filter the reports that contain "Host" in the report name:
+     *     icingacli reporting list --filter=*Host
+     */
+    public function indexAction()
+    {
+        $sort = strtolower($this->params->get('sort', 'id'));
+
+        if ($sort !== 'id' && $sort !== 'name' && $sort !== 'author') {
+            throw new InvalidArgumentException(sprintf('Sorting by %s is not supported', $sort));
+        }
+
+        $direction = $this->params->get('direction', 'ASC');
+
+        $reports = Model\Report::on($this->getDb());
+        $reports
+            ->with(['reportlets'])
+            ->orderBy($sort, $direction);
+
+        $filter = $this->params->get('filter');
+        if ($filter !== null) {
+            $reports->filter(Filter::like('name', $filter . '*'));
```

Comment:
Web automatically inserts asterisks before and after the filter value. We should do the same here. But to give the user some control, I would add the asterisks automatically only if the value **not** already contains an asterisk.